examples/training_example.py

```
@@ -0,0 +1,317 @@
+                if attempt < 4:
+                    print(f"  Retry {attempt + 1}/5 after error: {e}")
+                else:
+                    raise
+
+        print(f"Loss after loading checkpoint: {loss_after_load:.4f}")
+
+        # Verify the checkpoint restored the weights correctly
+        if abs(loss_after_load - loss_before_changes) < 0.01:
+            print("✓ R2 checkpoint round-trip PASSED - weights restored correctly!")
+        else:
+            print(f"✗ R2 checkpoint round-trip FAILED - loss mismatch")
```

Ruff (0.14.8)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)